docs/source/extending.rst

```diff
@@ -0,0 +1,103 @@
+.. _extending:
+
+
+.. note::
+
+   Make sure you've read the `design section <design>`_ of the documentation,
+   otherwise the rationale behind extending ibis won't make much sense.
+
+
+Extending Ibis
+==============
+
+There are two integration points that typical users want to extend:
+
+#. Additional operations
+#. Support for a new backend
+
+Adding a new operation (``Node`` subclass)
+------------------------------------------
+
+Let's go through adding a `sha1`_ method to ibis, implemented in the BigQuery
+backend.
+
+Step 1: Define the operation (Node)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The `SHA1
+<https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#sha1>`_
+function in BigQuery takes a string or bytes and returns a bytestring of length
+20.
+
+.. code-block:: python
```

Comment:
you could do these as ipython blocks (I think) esp if you want to use them later

Reply:
I'm actually going to move this into a notebook